src/libsyntax/ast.rs

```diff
@@ -1,33 +1,96 @@
-// The Rust abstract syntax tree.
+//! The Rust abstract syntax tree module.
+//!
+//! This module contains common structures forming the language AST.
+//! Two main entities in the module are [`Item`] (which represents an AST element with
+//! additional metadata), and [`ItemKind`] (which represents a concrete type and contains
+//! information specific to the type of the item).
+//!
+//! Other module items that worth mentioning:
+//! - [`TyKind`]: The various kinds of type recognized by the compiler.
```

Comment:
```suggestion
//! - [`Ty`] and [`TyKind`]: A parsed Rust type.
```

(also prefer Rust instead of `rust` everywhere)